### DISCUSSION TITLE
SourceController Service purpose

### DISCUSSION THREAD

**Discussion by rollandf** [fluxcd/flux2#3325] [Q&A] [answered]
Hi,

What is the purpose of the SourceController [Service ](https://github.com/fluxcd/source-controller/blob/main/config/manager/service.yaml)?

Is it needed for use case of only Helm handling with only sourcecontroller and helmcontroller?

Thanks!

**Comment by stefanprodan** [comment, marked as answer]
Flux source-controller fetches the Helm charts from upstream (Helm, Git, OCI repos) and stored them in its local cache, then helm-controller uses the source-controller cache to perform install and upgrades, without that ClusterIP service nothing works. The source-controller ClusterIP service is used by helm-controller to access the cache.

**Reply by rollandf** [reply]
Thanks!
